lib/remote-storage.js

```diff
@@ -71,8 +74,26 @@ module.exports = class RemoteStorage {
     }
     this.bucket = creds.params.Bucket
 
+    // Configure proxy support for AWS SDK v3
+    // Check for proxy configuration from environment variables (supporting both cases)
+    const httpsProxy = process.env.https_proxy || process.env.HTTPS_PROXY
+    const httpProxy = process.env.http_proxy || process.env.HTTP_PROXY
+
+    // Priority: HTTPS proxy > HTTP proxy (HTTPS takes precedence for S3)
+    const proxyUrl = httpsProxy || httpProxy
+
+    // Configure S3 client - only add requestHandler if proxy is configured
+    const s3Config = { credentials, region }
+    if (proxyUrl) {
+      const agent = new ProxyAgent(proxyUrl)
```

Comment:
just do this: 
`const agent = new ProxyAgent()`

[proxy-agent](https://www.npmjs.com/package/proxy-agent) module uses [proxy-from-env](https://www.npmjs.com/package/proxy-from-env) which automatically uses the proxy env variables.